### DISCUSSION TITLE
Setup style

### DISCUSSION THREAD

**Discussion by bodograumann** [DamianOsipiuk/vue-query#35] [Q&A]
Hi,

I am just looking at this library and it looks quite promising.
Would have expected it to be registered like other vue-plugins though.
Is there a reason you don’t use the following?
```
app.use(useQueryProvider())
```

**Comment by DamianOsipiuk** [comment]
Hi,
There is no particular reason for that. We could expose the registration logic as a plugin.
This would be a non-breaking change but just another method of using the library.

The only thing that would need to be discovered here would be the differences between plugin registration in Vue 2 and Vue 3.
I'm currently a bit out of time, so feel free to open a PR.

**Reply by wobsoriano** [reply]
I prefer the current approach tho. I can just import it in nuxt easily
